### ISSUE TITLE
Is it possible use apiserver address replace etcd_endpoints in calico-config configmap

### ISSUE THREAD

**Issue by yaowenying** [projectcalico/calico#4730]
At CM config  : 
cni_network_config-> etcd_endpoints
etcd_endpoints
the param will make all calico/node connect the etcd ,it is bad if etcd maintance like add node and so on.

## Expected Behavior

Is there any way use apiserver parameters replace the etcd_endpoints.

**Comment by lwr20** [comment]
If you're just going to use the kubernetes etcd as the calico datastore, just use the kubernetes API datastore mode:

https://docs.projectcalico.org/getting-started/kubernetes/self-managed-onprem/onpremises#install-calico-with-kubernetes-api-datastore-50-nodes-or-less

**Comment by yaowenying** [comment]
> If you're going to use the kubernetes etcd as the calico datastore, just use the kubernetes API datastore mode:
> 
> https://docs.projectcalico.org/getting-started/kubernetes/self-managed-onprem/onpremises#install-calico-with-kubernetes-api-datastore-50-nodes-or-less

Hello， i hava used the apiserver as datastore in new k8s cluster，but there is any way about the old cluster upgrade from etcd to apiserver。

**Comment by lwr20** [comment]
> but there is any way about the old cluster upgrade from etcd to apiserver

Yes, see: https://docs.projectcalico.org/maintenance/datastore-migration
